iothub/service/src/IotHubClientWebSocket.cs

```diff
@@ -229,24 +229,19 @@ public async Task<int> ReceiveAsync(byte[] buffer, int offset, int size, TimeSpa
                 {
                     // Ignore pong frame and start over
                     totalBytesRead = 0;
-                    do
-                    {
-                        bytesRead = await WebSocketStream.ReadAsync(header, totalBytesRead, header.Length - totalBytesRead).ConfigureAwait(false);
-                        if (bytesRead == 0)
-                        {
-                            throw new IOException(FramingPrematureEOF, new InvalidDataException("IotHubClientWebSocket was expecting more bytes"));
-                        }
+                    totalBytesRead = await ReadFromStreamAsync(WebSocketStream, header).ConfigureAwait(false);
```

Comment:
Since there are multiple instances of this warning across this class and I didn't want to clutter the class with #if statements, I created methods that would contain the framework conditions in there.